### PR TITLE
:warning: Rename TagType to TagCategory

### PR DESCRIPTION
**Pull request by mansam**
Signed-off-by: Sam Lucidi <slucidi@redhat.com>

### CODE REVIEW

File: api/tagcategory.go [hunk 1]
```diff
@@ -11,70 +11,70 @@ import (
...
-	resource := TagType{}
+	resource := TagCategory{}
 	resource.With(m)
 	ctx.JSON(http.StatusOK, resource)
 }
 
 // List godoc
-// @summary List all tag types.
-// @description List all tag types.
+// @summary List all tag categorys.
```
Comment (jortel) [inline api/tagcategory.go:61]: Simple search and replace error. categories

File: migration/v3/model/application.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+	Repository        JSON
+	Binary            string
+	Facts             JSON
+	Comments          string
+	Tasks             []Task     `gorm:"constraint:OnDelete:CASCADE"`
+	Tags              []Tag      `gorm:"many2many:ApplicationTags;constraint:OnDelete:CASCADE"`
+	Identities        []Identity `gorm:"many2many:ApplicationIdentity;constraint:OnDelete:CASCADE"`
+	BusinessServiceID *uint      `gorm:"index"`
+	BusinessService   *BusinessService
+}
```
Comment (jortel) [inline migration/v3/model/application.go:18]: What changed in the application model?

File: addon/tag.go [hunk 3]
```diff
@@ -97,8 +97,8 @@ type TagType struct {
 
 //
 // Create a tag-type.
-func (h *TagType) Create(m *api.TagType) (err error) {
-	err = h.client.Post(api.TagTypesRoot, m)
+func (h *TagType) Create(m *api.TagCategory) (err error) {
```
Comment (jortel) [inline addon/tag.go:100]: Let's rename the struct to TagCategory.

**Comment by jortel** [comment]
This looks good!
Just a few comments/questions.

File: addon/tag.go (hunk 3 above)
Comment (mansam) [reply addon/tag.go:100]: Ah, missed a spot.

File: migration/v3/model/application.go (hunk 2 above)
Comment (mansam) [reply migration/v3/model/application.go:18]: The Tags relationship is now to the v3 tag.

**Review by aufi — COMMENTED** [review]
Watching the change in order to update export/import CLI as needed.

**Comment by jortel** [comment]
Please rename this the `tp` argument to `category` [here](https://github.com/konveyor/tackle2-hub/blob/f4931ad5085c949bf47524743c78d59ac847a487/addon/tag.go#L60) on Find().

**Review by jortel — APPROVED** [review]
LGTM

**Pull request merged**